app/main/posts/views/share/post-export.directive.js

```diff
@@ -14,74 +14,29 @@ function PostExportDirective() {
 
 PostExportController.$inject = [
     '$scope',
-    '$translate',
-    'PostEndpoint',
-    'ConfigEndpoint',
     'Notify',
-    '$q',
     'PostFilters',
     '_',
-    '$window',
-    '$location',
-    '$timeout'
+    'DataExport'
 ];
 function PostExportController(
     $scope,
-    $translate,
-    PostEndpoint,
-    ConfigEndpoint,
     Notify,
-    $q,
     PostFilters,
     _,
-    $window,
-    $location,
-    $timeout
+    DataExport
 ) {
-    $scope.loading = false;
     $scope.getQuery = getQuery;
-    $scope.prepareExport = prepareExport;
-    $scope.showCSVResults = showCSVResults;
-    $scope.requestExport = requestExport;
-    $scope.loadingStatus = loadingStatus;
 
     $scope.exportPostsConfirmation = function () {
         /**
          * Trigger confirm notification for user.
          * When the user accepts, get the CSV
          */
         Notify.confirm('notify.post.export').then(function (message) {
-            prepareExport();
+            DataExport.startExport(getQuery());
```

Comment:
Should we have a showProgress here too? Also, why is this in the directive but everything else is in the controller?

Reply:
Its controlled in the service so no need to care about it here

Reply:
This is the directive for the share-button that exports filtered data to the user, so a bit different than the data-export.controller that is connected to the export-module in the settings, thats why they are in different places. But they use the same service so have reused code as much as I could.